Point-of-No-Return/Character.h

```diff
@@ -16,20 +25,79 @@ class Character :public Object
 class Hero :public Character
 {
 public:
+	
+/**
+* @brief 主人公の前の座標を更新する
+* @detail キャラクターを移動させる前にcallする
+*/
+	void UpdatePreviousVertex();
+
+	// void HitBlock(std::vector<Vec2> vec2);
+
 	Hero(const std::string& name):Character(name)
 	{
+		
 		Initialize();
 	}
+	
 
 	void Update();
 
+/**
+* @brief 座標を修正する関数
+* @param 座標修正する方向
+* @param 当たっているブロックの座標
+* @param イテレータ
+* @detail HitCheckの中で呼び出す
+*/
+	void CorrectCoodinate(Direction correction, std::vector<Vec2> vec2, int i);
+
+/**
+* @brief previousをとってくる関数
+*/
+	inline Vec2 GetPrevious()
+	{
+		return previous;
+	};
+
 private:
+	/**
+	* @brief 主人公の初期化
+	* @detail コンストラクタで呼び出す
+	*/
 	void Initialize();
 
+	//コピーコンストラクタと演算子オーバーロードの隠蔽
 	Hero(const Hero&);
 	Hero& operator=(const Hero&);
 	Hero(const Hero&&);
 	Hero& operator=(const Hero&&);
+
+protected:
+	//! 前の座標を保存する変数
+	Vec2 previous = {0,0};
+	//あとでprimitive型wrapします
```

Comment:
```suggestion
	// TODO: あとでprimitive型wrapします
```
とかにしておきましょう。

VSのタスク一覧ウィンドウでTODOコメントを確認する癖を付けましょう。
タスク一覧ウィンドウに列挙される内容は、「ツール」→「オプション」→「環境」→「タスク一覧」から設定できるのでチーム内で合わせておきましょう。

Reply:
f4d645d で修正しておきましたので、タスク一覧ウィンドウでご確認お願いします。